src/main/scala/org/apache/spark/shuffle/ucx/CommonUcxShuffleManager.scala

```diff
@@ -34,20 +39,26 @@ abstract class CommonUcxShuffleManager(val conf: SparkConf, isDriver: Boolean) e
 
   private val setupThread = ThreadUtils.newDaemonSingleThreadExecutor("UcxTransportSetupThread")
 
+  logInfo("LEO CommonUcxShuffleManager")
+
+  def getTransport(): UcxShuffleTransport = { ucxTransport }
+  def setTransport(transport: UcxShuffleTransport): Unit = { ucxTransport = transport }
+
   setupThread.submit(new Runnable {
     override def run(): Unit = {
-      while (SparkEnv.get == null) {
-        Thread.sleep(10)
-      }
+      CommonUtils.safePolling(() => {},
+      () => {SparkEnv.get == null}, 10*1000,
```

Comment:
Why not just 10000?

Reply:
I thought that it will be easier to understand that it's 10 seconds. Anyway I took a different approach see latest changes.